internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -676,6 +677,34 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"bootstrap_profile": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"artifact_source": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(managedclusters.ArtifactSourceCache),
+								string(managedclusters.ArtifactSourceDirect),
+							}, false),
```

Comment:
We can make use of the `PossibleValuesFor...` helper function here

```suggestion
							ValidateFunc: validation.StringInSlice(managedclusters.PossibleValuesForArtifactSource(), false),
```